### PR TITLE
serverutil(engine): don't panic when notifier is closed in executor watcher

### LINKED ISSUE #6875: job manager panic unexpectedly
### What did you do?

Two bugs:
- job manager runs with some error, the server master will panic.
- If job manager wait group exits, the job manager doesn't exit totally and hangs.

### What did you expect to see?

_No response_

### What did you see instead?

```go
[2022/08/24 14:39:50.278 +08:00] [WARN] [server.go:894] ["Polling JobManager failed"] [error="[DFLOW:ErrMetaEntryNotFound]meta entry not found: record not found"] [errorVerbose="[DFLOW:ErrMetaEntryNotFound]meta entry not found: record not found\ngithub.com/pingcap/errors.AddStack\n\tgithub.com/pingcap/errors@v0.11.5-0.20211224045212-9687c2b0f87c/errors.go:174\ngithub.com/pingcap/errors.Trace\n\tgithub.com/pingcap/errors@v0.11.5-0.20211224045212-9687c2b0f87c/juju_adaptor.go:15\ngithub.com/pingcap/tiflow/engine/framework.(*DefaultBaseMaster).Poll\n\tgithub.com/pingcap/tiflow/engine/framework/master.go:441\ngithub.com/pingcap/tiflow/engine/servermaster.(*Server).runLeaderService.func12\n\tgithub.com/pingcap/tiflow/engine/servermaster/server.go:893\ngolang.org/x/sync/errgroup.(*Group).Go.func1\n\tgolang.org/x/sync@v0.0.0-20220722155255-886fb9371eb4/errgroup/errgroup.go:75\nruntime.goexit\n\truntime/asm_amd64.s:1571"]
[2022/08/24 14:39:50.278 +08:00] [INFO] [executor_manager.go:265] ["check executor alive finished"]
[2022/08/24 14:39:50.278 +08:00] [INFO] [gc_coordinator.go:67] ["default gc coordinator exited"]
[2022/08/24 14:39:50.279 +08:00] [INFO] [gc_runner.go:67] ["default gc runner exited"]
[2022/08/24 14:39:50.279 +08:00] [INFO] [server.go:876] ["executor manager exited"]
[2022/08/24 14:39:50.279 +08:00] [PANIC] [watch_executors.go:106] ["unknown ExecutorStatusChange type"] [change="{\"ID\":\"\",\"Tp\":\"\",\"Addr\":\"\"}"] [stack="github.com/pingcap/tiflow/engine/servermaster/serverutil.WatchExecutors\n\tgithub.com/pingcap/tiflow/engine/servermaster/serverutil/watch_executors.go:106\ngithub.com/pingcap/tiflow/engine/servermaster.(*Server).runLeaderService.func13\n\tgithub.com/pingcap/tiflow/engine/servermaster/server.go:904\ngolang.org/x/sync/errgroup.(*Group).Go.func1\n\tgolang.org/x/sync@v0.0.0-20220722155255-886fb9371eb4/errgroup/errgroup.go:75"]
```

### Versions of the cluster

Dataflow Engine version (run `tiflow version`):

master@https://github.com/pingcap/tiflow/commit/c7f2901769be96dc6c9ab6dc00df42238d3240df

### PR DESCRIPTION
**Pull request by amyangfei**
### What problem does this PR solve?

Issue Number: close #6875

### What is changed and how it works?

- Use correct context when starting `serverutil2.WatchExecutors` goroutine
- Return directly instead of panic when executor watcher returns an empty change result, which means the notifier has been closed.

#### Tests 

 - Unit test
 - Integration test

### Release note 

```release-note
None
```

### CODE REVIEW

File: engine/servermaster/serverutil/watch_executors.go [hunk 1]
```diff
@@ -77,6 +79,11 @@ func WatchExecutors(ctx context.Context, watcher executorWatcher, user executorI
 		case change = <-updates.C:
 		}
 
+		if change == emptyChange {
+			log.Info("receive empty executor change")
+			return cerrors.ErrExecutorWatcherClosed.GenWithStackByArgs()
+		}
```
Comment (gozssky) [inline engine/servermaster/serverutil/watch_executors.go:85]: Can we change `case change = <-updates.C` to `case change, ok = <-updates.C`?

Comment (amyangfei) [reply engine/servermaster/serverutil/watch_executors.go:86]: addressed, this way is better.

**Review by gozssky — APPROVED** [review]

**Review by maxshuang — APPROVED** [review]

**Pull request merged**